### PR TITLE
Feature: add anonymous donation block

### PR DESCRIPTION
**Pull request by jonwaldstein**
## Description

This adds the Anonymous Donation block to our core blocks.

## Affects

- This new block adds a checkbox field that integrates with the core donate controller to update the anonymous property on the donation model
- Updated the list of disallowed field names in the field name validator to take in all model properties as well.
- Added migration step for v2 -> v3 form migration

## Visuals

Screenshot 2023-08-08 at 3 48 19 PM

Screenshot 2023-08-09 at 11 40 15 AM

## Testing Instructions

- Add an anonymous donation block to a v3 form.
- Interact with the settings
- Go to the front-end and check the box for anonymous
- Complete the donation and check the back-end to make sure it persists.
- You should also see a line on the confirmation page.

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205223433088672

### CODE REVIEW

**Comment by jonwaldstein** [comment]
@glaubersilva this is ready for review.

File: src/FormBuilder/ViewModels/FormBuilderViewModel.php [hunk 1]
```diff
@@ -115,4 +121,26 @@ public function getGateways(): array
 
         return array_values($builderPaymentGatewayData);
     }
+
+    /**
+     * @unreleased
+     */
```
Comment (JasonTheAdams) [inline src/FormBuilder/ViewModels/FormBuilderViewModel.php:127]: Please add a small description explaining what this is for. This is really helpful.

**Review by JasonTheAdams — APPROVED** [review]
Great work, @jonwaldstein! I just added one tiny request, which I think I'll just take care of as not to take up your time.

File: src/FormBuilder/ViewModels/FormBuilderViewModel.php (hunk 1 above)
Comment (JasonTheAdams) [reply src/FormBuilder/ViewModels/FormBuilderViewModel.php]: Resolved in c1b410c8

**Comment by kjohnson** [comment]
Merge conflicts resolved.

**Pull request merged**